src/backend/replication/walproposer.c

```diff
@@ -1621,20 +1628,42 @@ AdvancePollState(int i, uint32 events)
 				wk->feedback.flushLsn = truncateLsn;
 				wk->feedback.hs.ts = 0;
 
-				/*
-				 * We want our term to be highest and unique, so choose max
-				 * and +1 once we have majority.
-				 */
-				propTerm = Max(walkeeper[i].greet.term, propTerm);
+				++n_connected;
+				if (n_connected <= quorum)
+				{
+					/* We're still collecting terms from the majority. */
+					propTerm = Max(wk->greet.term, propTerm);
+
+					/* Quorum is acquried, prepare the vote request. */
+					if (n_connected == quorum)
+					{
+						propTerm++;
+						elog(LOG, "Proposer connected to quorum (%d) safekeepers, propTerm=" INT64_FORMAT, quorum, propTerm);
```

Comment:
Postgres's policy is to start log messages with lowercase letter (though we violate it in proposer code in some places)